app/services/subcases-service.js

```diff
@@ -128,4 +128,33 @@ export default Service.extend({
     return phases;
   },
 
+  /**
+   * Finds or creates a subcase of type TYPE in your publicationFlow.
+   *
+   * @param subcaseType
+   * @param publicationFlow
+   * @param title
+   * @param shortTitle
+   * @returns {Promise<*>}
+   */
+  async findOrCreateSubcaseFromTypeInPublicationFlow(requiredSubcaseType, publicationFlow, title, shortTitle) {
+    // Load the subcase (should be just one).
+    const subcases = await this.store.query('subcase', {
+      filter: {
+        type: {
+          id: requiredSubcaseType.id,
+        },
+        'publication-flow': {
+          id: await publicationFlow.get('id'),
+        },
+      },
+      sort: '-created',
+      include: 'type',
+    });
+    if (subcases.length > 0) {
+      return subcases.firstObject;
+    }
+    return this.createSubcaseForPublicationFlow(publicationFlow, requiredSubcaseType, shortTitle, title);
```

Comment:
deze is async dus awaiten

Reply:
goh ja ik had die al een niveautje lager ge-await dus kon geen kwaad dat er een promise terugkwam. Maar 't is wel duidelijker zo. - aangepast